src/scope_name.rs

```diff
@@ -69,21 +65,15 @@ impl NameComponent {
             inner: NameComponentInner::Interpolation(s),
         }
     }
-    pub(crate) fn ident(token: SyntaxToken) -> Self {
+    pub(crate) fn ident(ident: ast::Ident) -> Self {
         Self {
-            inner: NameComponentInner::SourceIdentifierToken(token),
-        }
-    }
-    pub(crate) fn punct(token: SyntaxToken) -> Self {
-        Self {
-            inner: NameComponentInner::SourcePunctuationToken(token),
+            inner: NameComponentInner::SourceIdentifierToken(ident),
         }
     }
 }
 
 #[derive(Debug)]
 pub(crate) enum NameComponentInner {
     Interpolation(&'static str),
-    SourceIdentifierToken(SyntaxToken),
-    SourcePunctuationToken(SyntaxToken),
+    SourceIdentifierToken(ast::Ident),
```

Comment:
This was a special case handling for "inlined" functions, for example:
```
function a() {};
a();
// minified:
(function(){})();
//            ^ this open paren has `name: a`
```

While yes, the swc parser does not give us that open paren token at all, a better solution is to just save the `name` as https://github.com/getsentry/symbolic/pull/698 does. The caller frame points directly to that open parenthesis (that is essentially how JS stack traces work). By looking at the caller frame we can cover this case in general instead of just the very special case of immediately invoked inlined functions.